File: C++/Easy/58. Length of Last Word.cpp
// https://leetcode.com/problems/length-of-last-word

#include <string>

class Solution {
public:

    // s: consists of words and spaces --> returns length of last word in the string
    // word: maximal substring consisting of non space characters only
    
    int lengthOfLastWord(string s) {
        int end = s.length() - 1;
        
        while (end >= 0 && s[end] == ' ') {
            end--;
        }
        
        int start = end;
        while (start >= 0 && s[start] != ' ') {
            start--;
        }

        return end - start;

    }
};

File: Python3/Easy/58. Length of Last Word.py
# https://leetcode.com/problems/length-of-last-word

class Solution:
    
    # s: string consisting of words and spaces --> return length of last word in s
    def lengthOfLastWord(self, s: str) -> int:
        end = len(s) - 1

        while end >= 0 and s[end] == ' ':
            end = end - 1
        start = end

        while start >= 0 and s[start] != ' ':
            start = start - 1

        return end - start
        
